docs/index-type-operator.md

```diff
@@ -0,0 +1,121 @@
+
+-- equivalent of `index<Person, "alive" | "age"> | index<Person2, "alive" | "age">`
+type idxType4 = index<Person | Person2, "alive" | "age"> -- Error message: Property '"age" | "alive"' does not exist on type 'Person | Person2'
+```
+
+In the circumstance that the indexee is a type class or table with an `__index` metamethod, the `__index` metamethod will *only* be invoked if the indexer is not found within the current scope:
+```luau
+local exampleClass = { Foo = "eight" }
+local exampleClass2 = setmetatable({ Foo = 8 }, { __index = exampleClass })
+local exampleClass3 = setmetatable({ Bar = "nine" }, { __index = exampleClass })
+
+type exampleTy2 = index<typeof(exampleClass2), "Foo"> -- exampleTy2 = number
+type exampleTy3 = index<typeof(exampleClass3), "Foo"> -- exampleTy3 = string
+```
+
+Implementation is straight forward: the type of the indexee will be determined (table, class, etc) -> search through the properties of the indexee and reduce to the corresponding type of the indexer if it exists; otherwise, reduce to an error. 
+
+## Drawbacks
+
+A drawback to this feature is the possible increase in the cost of maintenance. In the end, this RFC proposes adding another built-in type operators to the new type system. However, the addition of this feature may be worthwhile, as the `index` type operator is a useful type feature that:
+1. Alleviates the need to manually keep types in sync
+2. Provides a powerful way to access the properties of an object and perform various operations on it with other type operators
+3. Allows the community to write code with fewer errors and more safety
+
+## Alternatives
+
+An alternative design can be depicted from the example below:
+```luau
+type Person3 = {
+  age: number,
+  name: string,
+  alive: boolean,
+  job: string
+}
+
+local function edgeCase(p: Person)
+  type unknownType = index<typeof(p), "job">
+end
+```
+In our current design, the program simply fails to reduce (and throws an error). However, it is worth noting that `index<typeof(p), "job">` can also reduce to type `unknown` because the parameter `p` can be of type `Person` or `Person3` (since tables support width subtyping; hence, `Person3` is a subtype of `Person`):
+- If `p` is of type `Person`, `index<typeof(p), "job">` should reduce to an error. 
+- If `p` is of type `Person3`, `index<typeof(p), "job">` should reduce to type `string`.
+
+Because there are conflicting types for `p` depending on the run time, it is safest for the program to reduce to a type `unknown`. In this design, we would need a way to determine if the indexee can be different at runtime. We could determine this through the implementation of more table types, specifically exact and inexact table types. Then, the program will have two cases when an indexee does not contain the indexer type:
+1. If the indexee is an inexact table type, reduce to an `unknown` type.
+2. If the indexee is an exact table type, fail to reduce and throw an error.
+
```

Comment:
I would add a brief bit here mentioning that we can consider adding syntactic sugar for the `index` type operator in the future, with suggestions like `type["index"]` and `type.index`.